### ISSUE TITLE
Folder updating when download file

### ISSUE THREAD

**Issue by Djaler** [elementary/files#520]
If I open directory, that contains files in downloading state, this folder is constantly update. This cause to inability to rename some another file, for example.
https://youtu.be/5qszKp4dg3M

**Comment by jeremypw** [comment]
In your video, which file(s) is/are downloading?

**Comment by Djaler** [comment]
Lucifer. Not that I tried to rename, If you about this.

**Comment by jeremypw** [comment]
OK, thanks - I'll try to confirm - I am puzzled why all three thumbnails were being invalidated.

**Comment by jeremypw** [comment]
Doesn't happen if downloading file is an iso (no thumbnail).  I'll try with a video or something else that generates a thumbnail.

**Comment by Djaler** [comment]
If this is important - I download this files using Deluge torrent client

**Comment by jeremypw** [comment]
Maybe relevant.  I just tried copying a video into the folder that Files was displaying (using a different app) but there was no continual updating - the file was only thumbnailed after copy was complete.

**Comment by jeremypw** [comment]
Did you download the three video files in the same session?  It is possible that the torrent client said download was complete while it was still cached and being copied to the file.

**Comment by Djaler** [comment]
No. The rest of the files have been downloaded a very long time ago.

**Comment by jeremypw** [comment]
OK.  Can you point to a suitable (legal) source of .mkv videos to so I can try to reproduce more closely?

**Comment by Djaler** [comment]
Not sure if I can.

**Comment by jeremypw** [comment]
OK, I have now produced as similar effect using ffmpeg to convert MP4 video to MKV.  This causes the MKV file thumbnail to periodically invalidate and redisplay, which does end any ongoing rename operation of another file.

I cannot reproduce the invalidation of the thumbs other files (even MKV ones) that are not being changed however.

Not sure why this should be specific for MKV files but hopefully this will be enough for me to find a fix.  I cannot give it a high priority as it requires quite a specific set of circumstances.

**Comment by arshubham** [comment]
Also affects .avi files when downloading with uGet Download manager. When the files are downloading two files are formed *.avi and *.avia.aria2. I can rename anything, without stopping the download or waiting it to finish and sometimes it is not an option. Maybe a workaround can be that files can stop updating when renaming the files or right click menu is open (since sometimes due to updating the files won't enter rename mode) and update as soon as rename is done.

**Comment by Djaler** [comment]
Update: This also happens when download mp3 files

**Comment by Djaler** [comment]
Update: This happens when new mp3 files appears in directory. For example, I started a batch download. And when I tried to rename one from already downloaded files, it just reset.

**Comment by jeremypw** [comment]
This is unlikely to be fixed in the Gtk3 version but a different rename mechanism is used in the Gtk4 version which should fix this.
